pjmedia/src/pjmedia/sdp_neg.c

```diff
@@ -1545,6 +1571,106 @@ static pj_status_t create_answer( pj_pool_t *pool,
     return has_active ? PJ_SUCCESS : status;
 }
 
+/*
+ * Flag:
+ * 1: pt found
+ * 2: codec found
+ * Return (bitwise):
+ * 0: pt and codec not found
+ * 1: only pt found
+ * 2: only codec found
+ * 3: pt and codec found
+ */
+static unsigned pt_exists(const media_pt_map *sess_med_pt_map, unsigned pt, 
+                          const pj_str_t* codec)
+{
+    unsigned i = 0;
+    unsigned cnt = sess_med_pt_map->med_pt_map_cnt;
+    unsigned max_cnt = PJ_ARRAY_SIZE(sess_med_pt_map->med_pt_map);
+    unsigned retval = 0;
+    for (; i < cnt && i < max_cnt; ++i) {
+        unsigned tmp_ret = 0;
+
+        if (sess_med_pt_map->med_pt_map[i].pt == pt) {
+            tmp_ret |= 1;
+        }
+        if (pj_strcmp(&sess_med_pt_map->med_pt_map[i].codec, codec) == 0) {
+            tmp_ret |= 2;
+        }
+
+        if (tmp_ret == 3)
+            return tmp_ret;
+
+        if (tmp_ret > retval)
+            retval = tmp_ret;
+    }
+    return retval;
+}
+
+#if PJMEDIA_SDP_NEG_MAINTAIN_SESSION_PT
+
+static media_pt_map* get_media_pt_map(pjmedia_sdp_neg* neg, pj_str_t* med_type)
+{
+    unsigned i = 0;
+    unsigned max_cnt = PJ_ARRAY_SIZE(neg->sess_med_pt_map);
+    for (; i < neg->sess_med_pt_map_cnt && i < max_cnt; ++i) {
+        if (pj_strcmp(&neg->sess_med_pt_map[i].med_type, med_type) == 0) {
+            break;
+        }
+    }
+    if (i < max_cnt) {
+        if (i == neg->sess_med_pt_map_cnt) {
+            pj_strdup(neg->pool_neg, &neg->sess_med_pt_map[i].med_type, med_type);
+            neg->sess_med_pt_map_cnt++;
+        }
+        return &neg->sess_med_pt_map[i];
+    }
+    return NULL;
+}
+
+static void update_media_pt_map(const pjmedia_sdp_session* sess,
+                                pjmedia_sdp_neg* neg)
```

Comment:
minor/naming: 
- this function seems to work on the session level, so perhaps the `media` can be confusing?